test/org/sagebionetworks/bridge/models/ClientInfoTest.java

```diff
@@ -0,0 +1,141 @@
+package org.sagebionetworks.bridge.models;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+
+public class ClientInfoTest {
+
+    private static final String VALID_UA_1 = "Asthma/26 (Unknown iPhone; iPhone OS 9.1) BridgeSDK/4";
+    private static final String VALID_UA_2 = "Cardio Health/1 (Unknown iPhone; iPhone OS 9.0.2) BridgeSDK/4";
+    private static final String VALID_UA_3 = "Unknown Client/14 BridgeJavaSDK/10";
+    
+    private static final String INVALID_UA_1 = "Amazon Route 53 Health Check Service; ref:c97cd53f-2272-49d6-a8cd-3cd658d9d020; report http://amzn.to/1vsZADi";
+    private static final String INVALID_UA_2 = "Integration Tests (Linux/3.13.0-36-generic) BridgeJavaSDK/3";
+    private static final String INVALID_UA_3 = "Mozilla/5.0 (Windows NT 5.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2224.3 Safari/537.36";
+    private static final String INVALID_UA_4 = "Mozilla/5 (Windows NT 5.1) Safari/537"; // Purposefully very ambiguous; still fails.
+    
+    @Test
+    public void hashEquals() {
+        EqualsVerifier.forClass(ClientInfo.class).suppress(Warning.NONFINAL_FIELDS).allFieldsShouldBeUsed().verify();
+    }
```

Comment:
Don't need to suppress non-final fields, since all fields are (and should be) final.